src/ctapipe/image/modifications.py

```diff
@@ -100,6 +111,195 @@ def _smear_psf_randomly(
     return new_image
 
 
+class WaveformModifier(TelescopeComponent):
+    """
+    Component to add NSB noise to R1 waveforms (intended in principle to be
+    applied on MC simulations, to make them closer to real data)
+    The noise waveforms are read from a dedicated sim_telarray file, which
+    must be produced with the same telescope array configuration (and other
+    simulation settings) as the file to which the noise is to be added,
+    but containing only NSB noise (electronic noise should be switched off)
+
+    The number of simulated noise events per telescope in the NSB file must
+    be at least twice the number of waveforms ("nsb_level") from that file
+    that we want to add up. If the NSB file is produced with a level of 25%
+    of dark NSB, and we want to simulate 10x dark NSB, then nsb_level=40 (
+    =10/0.25) and the file must contain at least 80 events. This is to
+    guarantee that the different noise waveforms are not too correlated
+
+    This class can eventually use as NSB file a real data DL0 file which
+    contains pedestal events.
+
+    """
+
+    nsb_file = Path(
+        default_value=None,
+        help="Path to a dedicated NSB-only file (e.g. from sim_telarray)",
+    ).tag(config=True)
+
+    nsb_level = Int(
+        default_value=1,
+        help=(
+            "Number of random instances of the NSB waveforms from the "
+            "NSB file to be added up to the waveform"
+        ),
+    ).tag(config=True)
+
+    n_noise_realizations = Int(
+        default_value=100,
+        help=(
+            "Number of different realizations of the total NSB waveform to "
+            "be created per pixel"
+        ),
+    ).tag(config=True)
+
+    rng_seed = Int(default_value=1, help="Seed for the random number generator").tag(
+        config=True
+    )
+
+    nsb_event_types = List(
```

Comment:
EventTypeFilter already has a config option called `allowed_types`, so this is not needed here if the filter is constructed inside this class.  See comment below about the construction of the filter: